client/app/bundles/course/assessment/pages/AssessmentShow/AssessmentDetails.tsx

```diff
@@ -0,0 +1,115 @@
+import { TableCell, TableRow } from '@mui/material';
+import { AssessmentData } from 'types/course/assessment/assessments';
+
+import { TableContainer } from 'lib/components/core/table';
+import PersonalStartEndTime from 'lib/components/extensions/PersonalStartEndTime';
+import useTranslation from 'lib/hooks/useTranslation';
+
+import translations from '../../translations';
+
+interface AssessmentDetailsProps {
+  for: AssessmentData;
+}
+
+const AssessmentDetails = (props: AssessmentDetailsProps): JSX.Element => {
+  const { for: assessment } = props;
+  const { t } = useTranslation();
+
+  // TODO: Migrate to Tables API once it is ready
+  return (
+    <TableContainer dense variant="outlined">
+      <TableRow>
+        <TableCell variant="head">{t(translations.gradingMode)}</TableCell>
+
+        <TableCell>
+          {assessment.autograded
+            ? t(translations.autograded)
+            : t(translations.manuallyGraded)}
+        </TableCell>
+      </TableRow>
+
+      <TableRow>
+        <TableCell variant="head">{t(translations.baseExp)}</TableCell>
```

Comment:
Good catch! I was excluding `baseExp`, `timeBonusExp`, and `bonusEndAt` in `show.json.jbuilder` if the course is not gamified, but for some reasons instead of not rendering the rows, I coalesced them to `-` (hyphen).

I have added a condition to not render each of these rows if each of their values is undefined.